aptos-move/block-executor/src/view.rs

```diff
@@ -1534,6 +1541,21 @@ impl<'a, T: Transaction, S: TStateView<Key = T::Key>, X: Executable> TModuleView
             state_key,
         );
 
+        // Enforce feature gating V2 loader implementation: TModuleView is no longer used in
+        // V2 interfaces because we implement storage traits directly. Use a debug assert to
+        // panic in tests, adn invariant violation for non-debug builds.
+        if self.runtime_environment.vm_config().use_loader_v2 {
+            let msg =
+                "ModuleView trait should not be used when loader V2 implementation is enabled"
+                    .to_string();
+            let err = Err(
+                PartialVMError::new(StatusCode::UNKNOWN_INVARIANT_VIOLATION_ERROR)
+                    .with_message(msg),
+            );
+            debug_assert!(err.is_ok());
```

Comment:
how can err be ok?

Reply:
Used a debug assert to panic in tests.  But otherwise in prod, we cannot, so it is some sort of gating where we just return an invariant violation. Not ideal, maybe we just debug assert it?

This actually caused some panics in prop tests where I have not implement module storage yet, so it was useful.